internal/characters/candace/burst.go

```diff
@@ -101,24 +89,25 @@ func (c *char) burstInfuseFn(char *character.CharWrapper, src int) {
 		weapon.WeaponClassSword:
 		c.Core.Player.AddWeaponInfuse(
 			char.Index,
-			"candace-infuse",
+			"candace-q-infuse",
 			attributes.Hydro,
 			60,
 			true,
 			combat.AttackTagNormal, combat.AttackTagExtra, combat.AttackTagPlunge,
 		)
 	}
-	c.Core.Tasks.Add(func() { c.burstInfuseFn(char, src) }, 30)
+	c.QueueCharTask(func() { c.burstInfuseFn(char, src) }, 30)
 }
 
 func (c *char) burstSwap() {
 	c.Core.Events.Subscribe(event.OnCharacterSwap, func(args ...interface{}) bool {
-		if c.waveCount > 2 {
+		if !c.StatusIsActive(burstKey) {
 			return false
 		}
 		next := args[1].(int)
 		char := c.Core.Player.Chars()[next]
-		if !char.StatusIsActive(burstKey) {
+		c.burstInfuseFn(char, c.burstSrc)
```

Comment:
the burst infusion timer should be resetting on every swap so you need to also use a infuseSrc.

Reply:
This is covered in `burstInfuseFn()` by stopping if the active character changed. Context:
```go
func (c *char) burstInfuseFn(char *character.CharWrapper, src int) {
	if src != c.burstSrc {
		return
	}
	if c.Core.Player.Active() != char.Index {
		return
	}
```

Reply:
makes sense